pkg/cloud/aws/sqs/service.go

```diff
@@ -0,0 +1,280 @@
+package sqs
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/applike/gosoline/pkg/cfg"
+	"github.com/applike/gosoline/pkg/encoding/json"
+	"github.com/applike/gosoline/pkg/log"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/sqs"
+	"github.com/aws/aws-sdk-go-v2/service/sqs/types"
+)
+
+const DefaultVisibilityTimeout = "30"
+
+type ServiceSettings struct {
+	AutoCreate bool
+}
+
+//go:generate mockery -name Service
+type Service interface {
+	CreateQueue(ctx context.Context, settings *Settings) (*Properties, error)
+	QueueExists(ctx context.Context, name string) (bool, error)
+	GetPropertiesByName(ctx context.Context, name string) (*Properties, error)
+	GetPropertiesByArn(ctx context.Context, arn string) (*Properties, error)
+	GetUrl(ctx context.Context, name string) (string, error)
+	GetArn(ctx context.Context, url string) (string, error)
+	Purge(ctx context.Context, url string) error
+}
+
+type service struct {
+	lck      sync.Mutex
+	logger   log.Logger
+	client   Client
+	settings *ServiceSettings
+}
+
+func NewService(ctx context.Context, config cfg.Config, logger log.Logger, optFns ...ClientOption) (*service, error) {
+	var err error
+	var client Client
+
+	if client, err = ProvideClient(ctx, config, logger, "default", optFns...); err != nil {
+		return nil, fmt.Errorf("can not create client: %w", err)
+	}
+
+	settings := &ServiceSettings{
+		AutoCreate: config.GetBool("cloud.aws.sqs.autoCreate", false),
```

Comment:
I liked the defaults on the struct better - makes it a little clearer when looking at it

Reply:
changed